### PR TITLE
WT-3807 clang static analysis updates

### PR DESCRIPTION
**Pull request by keithbostic**
Bring the tree up-to-date with the latest clang release static analysis tools.

### CODE REVIEW

**Comment by keithbostic** [comment]
autobuild: retest this please

**Comment by keithbostic** [comment]
autobuild: retest this please

**Comment by keithbostic** [comment]
autobuild: retest this please

**Comment by keithbostic** [comment]
autobuild: retest this please

File: src/lsm/lsm_tree.c [hunk 1]
```diff
@@ -1378,9 +1378,9 @@ __wt_lsm_tree_worker(WT_SESSION_IMPL *session,
 	u_int i;
 	bool exclusive, locked, need_release;
 
-	locked = false;
-	need_release = false;
+	locked = need_release = false;
```
Comment (sulabhM) [inline src/lsm/lsm_tree.c]: false initialisation here seems redundant because both locked and need_release are set to true a few lines below.

Comment (keithbostic) [reply src/lsm/lsm_tree.c]: @sulabhM, agreed -- I'll send out email to explain my thinking.

File: src/include/wiredtiger.in [hunk 2]
```diff
@@ -2585,8 +2585,8 @@ struct __wt_connection {
  *
  * @param home The path to the database home directory.  See @ref home
  * for more information.
- * @param errhandler An error handler.  If <code>NULL</code>, a builtin error
- * handler is installed that writes error messages to stderr. See
+ * @param event_handler An event handler. If <code>NULL</code>, a default
+ * event handler is installed that writes error messages to stderr. See
  * @ref error_handling_event for more information.
```
[the remark this answers is not in the document]
Comment (keithbostic) [reply src/include/wiredtiger.in]: Thanks, @nehakhatri5, I agree with you.

I've tried to fix this in 6f71d19, please let me know if those changes are OK.

**Comment by nehakhatri5** [comment]
Thanks @keithbostic , LGTM.

**Pull request merged**